src/cloudevent/mocks/database/helpers.ts

```diff
@@ -0,0 +1,101 @@
+import { CloudFunction, database } from 'firebase-functions/v2';
+import { DeepPartial } from '../../types';
+import {
+  exampleDataSnapshot,
+  exampleDataSnapshotChange,
+} from '../../../providers/database';
+import { getBaseCloudEvent } from '../helpers';
+import { Change } from 'firebase-functions';
+
+export function getDatabaseSnapshotCloudEvent(
+  cloudFunction: CloudFunction<database.DatabaseEvent<any>>,
+  cloudEventPartial?: DeepPartial<database.DatabaseEvent<any>>
+) {
+  const {
+    instance,
+    firebaseDatabaseHost,
+    ref,
+    location,
+    params,
+  } = getCommonDatabaseFields(cloudFunction, cloudEventPartial);
+
+  const data =
+    (cloudEventPartial?.data as database.DataSnapshot) ||
+    exampleDataSnapshot(ref);
+
+  return {
+    // Spread common fields
+    ...getBaseCloudEvent(cloudFunction),
+
+    // Update fields specific to this CloudEvent
+    data,
+
+    instance,
+    firebaseDatabaseHost,
+    ref,
+    location,
+    params,
+  };
+}
+
+export function getDatabaseChangeSnapshotCloudEvent(
+  cloudFunction: CloudFunction<
+    database.DatabaseEvent<Change<database.DataSnapshot>>
+  >,
+  cloudEventPartial?: DeepPartial<
+    database.DatabaseEvent<Change<database.DataSnapshot>>
+  >
+): database.DatabaseEvent<Change<database.DataSnapshot>> {
+  const {
+    instance,
+    firebaseDatabaseHost,
+    ref,
+    location,
+    params,
+  } = getCommonDatabaseFields(cloudFunction, cloudEventPartial);
+
+  const data =
+    (cloudEventPartial?.data as Change<database.DataSnapshot>) ||
+    exampleDataSnapshotChange(ref);
+
+  return {
+    // Spread common fields
+    ...getBaseCloudEvent(cloudFunction),
+
+    // Update fields specific to this CloudEvent
+    data,
+
+    instance,
+    firebaseDatabaseHost,
+    ref,
+    location,
+    params,
+  };
+}
+
+export function getCommonDatabaseFields(
+  cloudFunction: CloudFunction<database.DatabaseEvent<any>>,
+  cloudEventPartial?: DeepPartial<database.DatabaseEvent<any>>
+) {
+  const instance =
+    (cloudEventPartial?.instance as string) ||
+    cloudFunction.__endpoint.eventTrigger.eventFilters.instance ||
```

Comment:
heads up, `instance` can also be in `eventFilterPathPatterns`. Just add
```
cloudFunction.__endpoint.eventTrigger.eventFilterPathPattern.instance ||
```

Reply:
Thanks for catching this!